include/assert.h

```diff
@@ -42,60 +42,68 @@
 #undef DEBUGASSERT  /* Like ASSERT, but only if CONFIG_DEBUG_ASSERTIONS is defined */
 #undef DEBUGVERIFY  /* Like VERIFY, but only if CONFIG_DEBUG_ASSERTIONS is defined */
 
-#if !defined(CONFIG_HAVE_FILENAME) || !defined(CONFIG_DEBUG_ASSERTIONS_FILENAME)
+/* Macro to define the assertions file name and file line
+ * | Function         |CONFIG                            | Show name/line |
+ * | ---              | ---                              | ---            |
+ * |assert(), ASSERT()|CONFIG_ASSERTIONS_FILENAME=y      | Yes            |
+ * |assert(), ASSERT()|CONFIG_ASSERTIONS_FILENAME=n      | No             |
+ * |DEBUGASSERT()     |CONFIG_DEBUG_ASSERTIONS_FILENAME=y| Yes            |
+ * |DEBUGASSERT()     |CONFIG_DEBUG_ASSERTIONS_FILENAME=n| No             |
+ */
+
+#ifdef CONFIG_HAVE_FILENAME
+#  ifdef CONFIG_DEBUG_ASSERTIONS_FILENAME
+#    define __DEBUG_ASSERT_FILE__ __FILE__
+#    define __DEBUG_ASSERT_LINE__ __LINE__
+#  endif
+#  ifdef CONFIG_ASSERTIONS_FILENAME
+#    define __ASSERT_FILE__ __FILE__
+#    define __ASSERT_LINE__ __LINE__
+#  endif
+#endif
+
+#ifndef __DEBUG_ASSERT_FILE__
+#  define __DEBUG_ASSERT_FILE__ 0
+#  define __DEBUG_ASSERT_LINE__ 0
+#endif
+
+#ifndef __ASSERT_FILE__
 #  define __ASSERT_FILE__ 0
 #  define __ASSERT_LINE__ 0
-#else
-#  define __ASSERT_FILE__ __FILE__
-#  define __ASSERT_LINE__ __LINE__
 #endif
 
-#define PANIC() __assert(__ASSERT_FILE__, __ASSERT_LINE__, "panic")
-#define PANIC_WITH_REGS(msg, regs) _assert(__ASSERT_FILE__, \
-                                           __ASSERT_LINE__, msg, regs)
+#define PANIC() __assert(__DEBUG_ASSERT_FILE__, __DEBUG_ASSERT_LINE__, "panic")
+#define PANIC_WITH_REGS(msg, regs) _assert(__DEBUG_ASSERT_FILE__, \
+                                           __DEBUG_ASSERT_LINE__, msg, regs)
```

Comment:
```suggestion
#define PANIC() __assert(__ASSERT_FILE__, __ASSERT_LINE__, "panic")
#define PANIC_WITH_REGS(msg, regs) _assert(__ASSERT_FILE__, \
                                           __ASSERT_LINE__, msg, regs)
```

How about `PANIC()` uses `__ASSERT_FILE__` and `DEBUGPANIC()` uses `__DEBUG_ASSERT_FILE__`?

Reply:
Addressed! Thank you!